### PR TITLE
Conditions allowable actions and states on present

### PR DESCRIPTION
**Pull request by ptoman**
Conditions the allowable actions & states on the present state/actions. Relies on having an implementation of POMDPs.state_index, POMDPs.action_index, and POMDPs.actions by client MDP (which might simply return all states).
This addresses #4.

### CODE REVIEW

**Comment by ptoman** [comment]
I can edit the test to implement the missing functions for the MDP instance that make the test fail (`No implementation of action_index{S,A} for problem`) -- but I'd be more comfortable if a repo owner checked the appropriate behavior and updated tests appropriately.

**Comment by mykelk** [comment]
Cool. @etotheipluspi and @zsunberg (and maybe others) will check this out.

**Comment by etotheipluspi** [comment]
Tests failed because the grid world problem didn't have the `action_index` implementation. `POMDPModels.jl` has been updated, and tests should be passing now.

File: src/vanilla.jl [hunk 1]
```diff
@@ -132,30 +132,33 @@ function solve(solver::ValueIterationSolver, mdp::Union{MDP,POMDP}, policy=creat
...
-        for (istate, s) in enumerate(iterator(sspace))
-            old_util = util[istate] # for residual 
+        for s in iterator(sspace)
+            sidx = state_index(mdp, s)
+            old_util = util[sidx] # for residual 
             max_util = -Inf
             # action loop
             # util(s) = max_a( R(s,a) + discount_factor * sum(T(s'|s,a)util(s') )
-            for (iaction, a) in enumerate(iterator(aspace))
+            sub_aspace = POMDPs.actions(mdp, s, aspace)
```
Comment (zsunberg) [inline src/vanilla.jl:141]: Where is sub_aspace used? I couldn't find it below. Should aspace be replaced with sub_aspace in 142

**Review by zsunberg — COMMENTED** [review]
Thanks a bunch Pam! this looks good. I think some of it (like the enumerate(iterator(sspace))) was left over from before we had finalized the interface, so thanks for catching that! I left one comment in the code - it looks like you might have forgotten to use sub_aspace

**Comment by zsunberg** [comment]
I noticed a mistake in the code unrelated to this and pushed the fix to master, so I think you will have to merge master into your branch before this can be merged

File: src/vanilla.jl (hunk 1 above)
Comment (ptoman) [reply src/vanilla.jl:141]: You definitely caught a copy & paste error there. I've fixed it in the decomposed_tests branch I'm working in, as well as adding more tests, and I'm able to push from local machine directly now, so copying & pasting errors shouldn't happen again in this repo.

**Comment by zsunberg** [comment]
Replaced this with #11

**Pull request closed without merging**